### PR TITLE
Validate path in external metric name

### PR DESCRIPTION
**Pull request by MaciekPytel**
Add validation following comments on https://github.com/kubernetes/kubernetes/pull/60220

### CODE REVIEW

**Comment by MaciekPytel** [comment]
cc: @DirectXMan12

File: pkg/apis/autoscaling/validation/validation.go [hunk 1]
```diff
@@ -197,6 +197,10 @@ func validateExternalSource(src *autoscaling.ExternalMetricSource, fldPath *fiel
 		allErrs = append(allErrs, field.Required(fldPath.Child("metricName"), "must specify a metric name"))
 	}
 
+	for _, msg := range pathvalidation.IsValidPathSegmentName(src.MetricName) {
```
Comment (liggitt) [inline pkg/apis/autoscaling/validation/validation.go]: can put this in an else block if `len(src.MetricName) > 0`

**Comment by liggitt** [comment]
adding to milestone so we don't release an unvalidated API field

File: pkg/apis/autoscaling/validation/validation.go (hunk 1 above)
Comment (MaciekPytel) [reply pkg/apis/autoscaling/validation/validation.go]: done

**Comment by MaciekPytel** [comment]
Rebased to fix merge conflicts.

**Pull request merged**